### PR TITLE
TINY-7588: Fixed new `prefer-fun` errors for `Fun.constant` usage

### PR DESCRIPTION
**Pull request by lnewson**
Related Ticket: TINY-7588

Description of Changes:
* Fix places where `Fun.constant` should have been used in existing code.

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/alloy/src/main/ts/ephox/alloy/api/component/Component.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ const build = (spec: ComponentDetail): AlloyComponent => {
     // INVESTIGATE: Not sure about how to handle text nodes here.
     const subs = Arr.bind(children, (child) => systemApi.get().getByDom(child).fold(
       () => [ ],
-      (c) => [ c ]
+      Arr.pure
```
Comment (lnewson) [inline modules/alloy/src/main/ts/ephox/alloy/api/component/Component.ts:89]: Accidentally left this in, but it also makes sense to use this 🤷‍♀️

**Review by ltrouton — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/util/ToolsTest.ts [hunk 2]
```diff
@@ -12,7 +12,9 @@ describe('browser.tinymce.core.util.ToolsTest', () => {
 
   context('create', () => {
     it('TINY-7358: Multiple calls to create should create different objects', () => {
+      // eslint-disable-next-line @tinymce/prefer-fun
```
Comment (ashfordneil) [inline modules/tinymce/src/core/test/ts/browser/util/ToolsTest.ts:15]: Why?

File: modules/tinymce/src/plugins/imagetools/main/ts/ui/Buttons.ts [hunk 3]
```diff
@@ -64,12 +64,11 @@ const register = (editor: Editor) => {
   editor.ui.registry.addContextMenu('imagetools', {
     update: (element) =>
       // since there's no menu item available, this has to be it's own thing
-      Actions.getEditableImage(editor, element).fold(() => [], (_) => [{
+      Actions.getEditableImage(editor, element).map((_) => ({
```
Comment (ashfordneil) [inline modules/tinymce/src/plugins/imagetools/main/ts/ui/Buttons.ts:67]:
```suggestion
      Actions.getEditableImage(editor, element).map(Fun.constant({
```

**Review by ashfordneil — APPROVED** [review]

File: modules/tinymce/src/core/test/ts/browser/util/ToolsTest.ts (hunk 2 above)
Comment (lnewson) [reply modules/tinymce/src/core/test/ts/browser/util/ToolsTest.ts:15]: This is a class constructor, it can't use constants.

**Pull request merged**